packages/backend-storage/src/construct.ts

```diff
@@ -146,15 +151,22 @@ export class AmplifyStorage
    * Store storage outputs using provided strategy
    */
   private storeOutput = (
-    outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput> = new StackMetadataBackendOutputStorageStrategy(
-      Stack.of(this)
-    )
+    outputStorageStrategy: BackendOutputStorageStrategy<{
+      version: '1';
+      payload: Record<string, string>;
+    }> = new StackMetadataBackendOutputStorageStrategy(Stack.of(this)),
+    isDefault: boolean = false
   ): void => {
+    /* The following code can guarantee there's only one `isDefault`
+     * because if we can only create one construct with the same `storageRegion` and `bucketName` name.
+     * The default bucket takes the `storageRegion` and `bucketName` name without a number post-fix.
+     */
+    const num = isDefault ? '' : Math.floor(Math.random() * 100);
     outputStorageStrategy.addBackendOutputEntry(storageOutputKey, {
       version: '1',
       payload: {
-        storageRegion: Stack.of(this).region,
-        bucketName: this.resources.bucket.bucketName,
+        [`storageRegion${num}`]: Stack.of(this).region,
+        [`bucketName${num}`]: this.resources.bucket.bucketName,
```

Comment:
cdk would throw an error saying "There's already a Construct name 'storageRegion'...", so we need to give it different names if it's not the default bucket.